diff --git a/bootstrap.conf b/bootstrap.conf
index 4553372..1dc8cbe 100644
--- a/bootstrap.conf
+++ b/bootstrap.conf
@@ -44,6 +44,7 @@ fseeko
 fstatat
 ftello
 futimens
+getdtablesize
 getopt-gnu
 gettime
 gitlog-to-changelog
diff --git a/src/safe.c b/src/safe.c
index 2aafbca..52a411b 100644
--- a/src/safe.c
+++ b/src/safe.c
@@ -21,7 +21,9 @@
 #include <config.h>
 #include <sys/types.h>
 #include <sys/stat.h>
-#include <sys/resource.h>
+#if !(defined _WIN32 && !defined __CYGWIN__)
+# include <sys/resource.h>
+#endif
 #include <sys/time.h>
 #include <fcntl.h>
 #include <stdio.h>
@@ -73,8 +75,8 @@ struct cached_dirfd {
 };
 
 static Hash_table *cached_dirfds;
-static rlim_t min_cached_fds = 8;
-static rlim_t max_cached_fds;
+static size_t min_cached_fds = 8;
+static size_t max_cached_fds;
 static LIST_HEAD (lru_list);
 
 static size_t hash_cached_dirfd (const void *entry, size_t table_size)
@@ -103,6 +105,7 @@ static void free_cached_dirfd (struct cached_dirfd *entry)
 
 static void init_dirfd_cache (void)
 {
+#if !(defined _WIN32 && !defined __CYGWIN__)
   struct rlimit nofile;
 
   if (getrlimit (RLIMIT_NOFILE, &nofile) == 0)
@@ -114,6 +117,9 @@ static void init_dirfd_cache (void)
     }
   else
     max_cached_fds = min_cached_fds;
+#else
+  max_cached_fds = getdtablesize ();
+#endif
 
   cached_dirfds = hash_initialize (min_cached_fds, nullptr,
 				   hash_cached_dirfd,
